src/luxonis_ml/loader/augmentations.py

```diff
@@ -334,4 +389,197 @@ def __init__(
+            border_value (int, optional): Padding value for images. Defaults to 0.
+            mask_value (int, optional): Padding value for masks. Defaults to 0.
+            always_apply (bool, optional): Defaults to False.
+            p (float, optional): Probability of applying the transform. Defaults to 1.0.
+        """
+        super().__init__(always_apply, p)
+
+        if not (0 <= border_value <= 255):
+            raise ValueError("Border value must be in range [0,255].")
+
+        if not (0 <= mask_value <= 255):
+            raise ValueError("Mask value must be in range [0,255].")
+
+        self.height = height
+        self.width = width
+        self.interpolation = interpolation
+        self.border_value = border_value
+        self.mask_value = mask_value
+
+    def update_params(self, params: Dict[str, Any], **kwargs) -> Dict[str, Any]:
+        params = super().update_params(params, **kwargs)
+
+        img_height = params["rows"]
+        img_width = params["cols"]
+
+        ratio = min(self.height / img_height, self.width / img_width)
+        new_height = int(img_height * ratio)
+        new_width = int(img_width * ratio)
+
+        # only supports center alignment
+        pad_top = (self.height - new_height) // 2
+        pad_bottom = pad_top
+
+        pad_left = (self.width - new_width) // 2
+        pad_right = pad_left
+
+        params.update(
+            {
+                "pad_top": pad_top,
+                "pad_bottom": pad_bottom,
+                "pad_left": pad_left,
+                "pad_right": pad_right,
+            }
+        )
+
+        return params
+
+    def apply(
+        self,
+        img: np.ndarray,
+        pad_top: int,
+        pad_bottom: int,
+        pad_left: int,
+        pad_right: int,
+        **params
+    ) -> np.ndarray:
+        resized_img = cv2.resize(
+            img,
+            (self.width - pad_left - pad_right, self.height - pad_top - pad_bottom),
+            interpolation=self.interpolation,
+        )
+        img_out = cv2.copyMakeBorder(
+            resized_img,
+            pad_top,
+            pad_bottom,
+            pad_left,
+            pad_right,
+            cv2.BORDER_CONSTANT,
+            self.border_value,
+        )
+        img_out = img_out.astype(img.dtype)
+        return img_out
+
+    def apply_to_mask(
+        self,
+        img: np.ndarray,
+        pad_top: int,
+        pad_bottom: int,
+        pad_left: int,
+        pad_right: int,
+        **params
+    ) -> np.ndarray:
+        resized_img = cv2.resize(
+            img,
+            (self.width - pad_left - pad_right, self.height - pad_top - pad_bottom),
+            interpolation=cv2.INTER_NEAREST,
+        )
+        img_out = cv2.copyMakeBorder(
+            resized_img,
+            pad_top,
+            pad_bottom,
+            pad_left,
+            pad_right,
+            cv2.BORDER_CONSTANT,
+            self.mask_value,
```

Comment:
Is 0 usually the background class by default?

Reply:
Yeah, `mask_value` is set to 0 by default